src/Trace/Tracer/ExtensionTracer.php

```diff
@@ -114,6 +124,43 @@ public function spans(): array
         }, opencensus_trace_list());
     }
 
+    /* This checks the numbet of spans in memory and if the count is more than the set limit, it exports all
+    the closed span present in memory, to free up the memory */
+    public function checkSpanLimit()
+    {
+        $count = count($this->spans());
+
+        // TODO:: to read from config
+        $limit = self::DEFAULT_SPAN_LIMIT;
+
+        if ($count > $limit) {
+
+            $closedSpans = [];
+            $spans = $this->spans();
+
+            foreach ($spans as $k) {
+                $endTime = $k->endTime();
+
+                if ($endTime->getTimestamp() != 0) {
+                    $closedSpans[] = $k;
+                }
+            }
+
+            $this->export($closedSpans);
```

Comment:
can we rename this to capture removal part of it too?

Reply:
done